pkg/controller/elasticsearch/nodespec/lifecycle_hook.go

```diff
@@ -165,7 +165,7 @@ ES_URL={{.ServiceURL}}
 
 log "retrieving node ID"
 retry 10 request -X GET "$ES_URL/_cat/nodes?full_id=true&h=id,name" $BASIC_AUTH
-if [ "$?"  -ne 0 ]; then
+if [ "$?" -ne 0 ]; then
 	error_exit "$status"
```

Comment:
I think `$status` needs to be declared outside the function `request` to be used globally, as here.

Reply:
You are right, I did not test that error condition.

Reply:
I found more issues while fixing this. But now we should be good.